engine/src/flutter/lib/ui/platform_dispatcher.dart

```diff
@@ -1506,6 +1506,88 @@ class PlatformDispatcher {
   // configurationId does not match any configuration.
   @Native<Double Function(Double, Int)>(symbol: 'PlatformConfigurationNativeApi::GetScaledFontSize')
   external static double _getScaledFontSize(double unscaledFontSize, int configurationId);
+
+  /// Color palette preferred by the operating system UI widgets.
+  ///
+  /// On platforms that do not support a UI color palette this value is null.
+  ///
+  /// As of the current release, system colors are supported on web only.
+  ///
+  /// This is typically used in conjunction with
+  /// [AccessibilityFeatures.highContrast]. In particular, on Windows, when
+  /// a user enables high-contrast mode, they may also pick specific colors that
+  /// should be used by application user interfaces.
+  ///
+  /// See also:
+  ///
+  ///   * https://drafts.csswg.org/css-color/#css-system-colors
+  ///   * https://developer.mozilla.org/en-US/docs/Web/CSS/system-color
+  ///   * https://developer.mozilla.org/en-US/docs/Web/CSS/@media/forced-colors
+  Map<String, SystemColor>? get systemColors => null;
```

Comment:
Thanks for adding the support on the engine side. I will think about how we integrate this with framework themes.